build/webpack.common.js

```diff
@@ -128,38 +131,54 @@ const withImageModuleLoader = returnModuleRuleWithConfig(
   ]
 );
 
-// exported helper function - returns common configuration, merged via object spread. Provided options take precedence
-const returnBasicConfigMergedWith = (customConfigurationForBuildMode = {}) => ({
-  entry: `${BOOTSTRAP_DIR}/index.js`,
-  output: {
-    path: BUILD_DIR,
-    publicPath: '',
-    filename: '[name].bundle.js',
-  },
-  module: {
-    // given the dynamic nature of rules and configuration, none a re provided by default. Use the exported `moduleLoaders` to return loaders with some default configs
-    rules: [],
-  },
-  // given different build modes will require different plugins/plugin configuration, none are provided by default. Use the exported `plugins` for commonly used plugins with default configuration
-  plugins: [],
-  resolve: {
-    alias: webpackAliases,
+const withTypescriptModuleLoader = returnModuleRuleWithConfig(
+  {
+    test: /\.tsx?$/,
+    exclude: /node_modules/,
   },
-  ...customConfigurationForBuildMode,
-});
+  ['ts-loader']
+);
+
+// exported helper function - returns common configuration, merged via object spread. Provided options take precedence
+const returnBasicConfigMergedWith = (customConfigurationForBuildMode = {}) =>
+  merge(
+    {
+      entry: `${BOOTSTRAP_DIR}/index.tsx`,
```

Comment:
It isn't merged yet, but when the server builds, it has `.ts` entry point - should/would this apply to client webpack build configs only? (Happy to change in my PR if this makes it in first)

Reply:
Absolutley - I wonder as the server and client will be different here is the merit in removing entry from common (as it now wont be)?